gms/gossiper.cc

```diff
@@ -762,9 +763,10 @@ future<gossiper::endpoint_permit> gossiper::lock_endpoint(inet_address ep) {
 // - failure_detector
 // - on_remove callbacks, e.g, storage_service -> access token_metadata
 void gossiper::run() {
-   // Run it in the background.
-  (void)seastar::with_semaphore(_callback_running, 1, [this] {
-    return seastar::async([this, g = this->shared_from_this()] {
```

Comment:
Why not do it in the timer itself? set_callback() accepts a scheduling group.

Reply:
OK. I will get back to this when I finish other higher priority issues.

Reply:
Done. Sent v2.